Build failure on sethrfore/r-srf/r version 4.0.3 (Mac OS X)

`R version 4.0.3 (2020-10-10) -- "Bunny-Wunnies Freak Out"` — Installed [like this](https://blog.theforce.top/2019/06/20/install-r-with-all-capabilities-and-openblas-openmp-on-macos-by-homebrew/)

```
> install.packages("ggplot2")
```

[snip...]

```
clang-11: warning: argument unused during compilation: '-L/usr/local/opt/llvm/lib' [-Wunused-command-line-argument]
unix/processx.c:46:28: error: 'TARGET_OS_IPHONE' is not defined, evaluates to 0 [-Werror,-Wundef-prefix=TARGET_OS_]
#if defined(__APPLE__) && !TARGET_OS_IPHONE
```

--- a/src/unix/processx.c
+++ b/src/unix/processx.c
@@ -43,9 +43,14 @@ void processx__create_connections(processx_handle_t *handle, SEXP private,
 #endif
 #endif
 
-#if defined(__APPLE__) && !TARGET_OS_IPHONE
+#if defined(__APPLE__)
+#include <TargetConditionals.h>
+#if ! TARGET_OS_IPHONE
 # include <crt_externs.h>
 # define environ (*_NSGetEnviron())
+#else
+extern char **environ;
+#endif
 #else
 extern char **environ;
 #endif

--- a/src/unix/processx.c
+++ b/src/unix/processx.c
@@ -44,13 +44,8 @@ void processx__create_connections(processx_handle_t *handle, SEXP private,
 #endif
 
 #if defined(__APPLE__)
-#include <TargetConditionals.h>
-#if ! TARGET_OS_IPHONE
 # include <crt_externs.h>
 # define environ (*_NSGetEnviron())
-#else
-extern char **environ;
-#endif
 #else
 extern char **environ;
 #endif

